datafusion/sql/src/statement.rs

```diff
@@ -866,12 +866,16 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             FileType::from_str(&file_type).map_err(|_| {
                 DataFusionError::Configuration(format!("Unknown FileType {}", file_type))
             })?
+        } else if let Some(format) = options.remove("format") {
+            // try to infer file format from the "format" key in options
+            FileType::from_str(&format)
+                .map_err(|e| DataFusionError::Configuration(format!("{}", e)))?
         } else {
             let e = || {
                 DataFusionError::Configuration(
-                "Format not explicitly set and unable to get file extension! Use STORED AS to define file format."
-                    .to_string(),
-            )
+                    "Format not explicitly set and unable to get file extension! Use STORED AS to define file format."
+                        .to_string(),
+                )
```

Comment:
One thing to note is that, currently, order of priority for resolving the `file_type` is:
1. `STORED AS`
2. Legacy: `COPY {} TO {} OPTIONS (format <>)`
3. File extension

Let me know if we care about the order of resolution & if this requires any change.

Reply:
I agree that this is the most sensible order to determine the file type 👍